instrumentation/net/http/handler.go

```diff
@@ -170,13 +170,16 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		},
 	})
 
+	labeler := &Labeler{}
+	ctx = injectLabeler(ctx, labeler)
+
 	h.handler.ServeHTTP(w, r.WithContext(ctx))
 
 	setAfterServeAttributes(span, bw.read, rww.written, rww.statusCode, bw.err, rww.err)
 
 	// Add request metrics
 
-	labels := semconv.HTTPServerMetricAttributesFromHTTPRequest(h.operation, r)
+	labels := append(labeler.labels, semconv.HTTPServerMetricAttributesFromHTTPRequest(h.operation, r)...)
```

Comment:
With the addition of the mutex a getter method is needed to ensure reads like this are concurrency safe.

Reply:
This would also need to return a copy of the labels then, wouldn't it?  I wonder whether it's sufficient to declare that the `Labeler` is not safe for use after the wrapped `http.Handler` has returned.  Maybe a getter that sets a flag that `Add()` checks before allowing the labels to be modified?

Reply:
I think returning a copy and making it clear in the documentation that this is the behavior would be sufficient.

Reply:
I don't think a copy is needed, as long as the labeler itself only appends and promises not to mutate the slice or the values it contains. It's not a big deal to copy, though.